scylla/src/transport/load_balancing/default.rs

```diff
@@ -1234,6 +1307,54 @@ mod tests {
                     .group([B, C, E]) // remote nodes
                     .build(),
             },
+            // Keyspace SS with RF=2 with enabled DC failover and rack-awareness
+            Test {
+                policy: DefaultPolicy {
+                    preferred_datacenter: Some("eu".to_owned()),
+                    preferred_rack: Some("r2".to_owned()),
+                    is_token_aware: true,
+                    permit_dc_failover: true,
+                    ..Default::default()
+                },
+                routing_info: RoutingInfo {
+                    token: Some(Token { value: 760 }),
+                    keyspace: Some(KEYSPACE_SS_RF_2),
+                    consistency: Consistency::One,
+                    ..Default::default()
+                },
+                // going though the ring, we get order: F , A , C , D , G , B , E
+                //                                      us  eu  eu  us  eu  eu  us
+                //                                      r2  r1  r1  r1  r2  r1  r1
+                expected_groups: ExpectedGroupsBuilder::new()
+                    .group([G]) // pick local rack replicas
```

Comment:
Having a single element rack group fails to test shuffling.

Reply:
Please add a test, that tests shuffling of local-rack replicas.

Reply:
I think it does test shuffling.
https://github.com/scylladb/scylla-rust-driver/blob/758ec3e17e248a77ec7d269d9c2e9a3d82a402c3/scylla/src/transport/load_balancing/default.rs#L707-L715

After some thought though, I realized that it would be quite difficult to test if returned local-rack replicas as shuffled. It is because local-rack replicas are always first in the plan (if available), an hence the first element from the plan is a picked local-rack replica. If `pick` chooses nodes in a random way but `fallback` doesn't, the returned group in plan is not deterministic anyway. Our testing framework has no way of detecting that a group is only partially shuffled.

I don't know if adding new functionality to the testing framework is worth it. I guess that I'm withdrawing my request for a test that checks shuffling of local-rack replicas. @wprzytula any thoughts on this?

Reply:
I think it would be nice to at least verify that `pick` chooses rack-local replicas randomly. Having a test with the "local rack replicas" group of size 2 should check this.

Later we can consider adjusting the tests - for example, in addition to checking plans that combine `pick` and `fallback` we could also check plans returned by `fallback` only.